tests/integration/test_pip.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from pathlib import Path
+
+import pytest
+
+from . import utils
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "test_params",
+    [
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/cachito-pip-without-deps.git",
+                ref="3fe2fc3cb8ffa36317cacbd9d356e35e17af2824",
+                packages=({"path": ".", "type": "pip"},),
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="All dependencies fetched successfully",
+            ),
+            id="pip_without_deps",
+        ),
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/cachito-pip-with-deps.git",
+                ref="56efa5f7eb4ff1b7ea1409dbad76f5bb378291e6",
+                packages=({"path": ".", "type": "pip"},),
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="All dependencies fetched successfully",
+            ),
+            id="pip_with_deps",
+        ),
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/cachito-pip-multiple.git",
+                ref="93c6c44b36075454a509d595850b81be29e53db0",
+                packages=(  # type: ignore
```

Comment:
Mypy fails here with:
```
tests/integration/test_pip.py:40: error: Argument "packages" to "TestParameters" has incompatible type "Tuple[Dict[Any, Any], Dict[str, str]]"; expected "Tuple[Dict[Any, Any]]"  [arg-type]
```

Only way I could make it stop was with the ignore comment :disappointed:

Reply:
Ah, it should be annotated as like this on the TestParameters class
```python
packages: tuple[dict[str, Any], ...]
```

Other options:
```python
packages: list[dict[str, Any]]  # and use lists instead
packages: Sequence[dict[str, Any]]  # and use whatever you want, but it's not 100% going to be json-able
```

Reply:
Without the ellipsis (...), a `tuple[T]` means a one-element tuple.

`tuple[T, U]` is a two-element tuple etc.